### PR TITLE
v1beta1 API for cosigned

### LINKED ISSUE #1835: [cosigned] Bump API version to v1beta1
**Description**

This issue contains a list of other issues needed to bump the current API version from v1alpha1 to v1beta1:

[x]  Image pattern selector: keep glob and remove regex  https://github.com/sigstore/cosign/issues/1834

cc @mattmoor @vaikas @coyote240 @DennyHoang @adamd-vmw @elfotografo007

### PR DESCRIPTION
**Pull request by vaikas**
#### Summary
Add v1beta1 API for cosigned.

#### Ticket Link
Fixes #1835 

-->
Fixes

#### Release Note

```release-note
Add v1beta1 API for cosigned
```

### CODE REVIEW

**Comment by vaikas** [comment]
@hectorj2f yeah, I need to figure out how to do this safely with the codegen :)

**Comment by vaikas** [comment]
I'm looking at the test failures, but it seems to not be able to get the latest release tag for scaffolding:
```
++ curl -s -u username:*** https://api.github.com/repos/sigstore/scaffolding/releases/latest
+ tag=null
+ echo 'Installing kind and knative using release'
+ curl -Lo ./setup-kind.sh https://github.com/sigstore/scaffolding/releases/download/null/setup-kind.sh

```
Notice the tag is `null`

File: pkg/apis/cosigned/v1beta1/clusterimagepolicy_types.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	// not even get evaluated, as the Policy is considered failed.
+	// +optional
+	Policy *Policy `json:"policy,omitempty"`
+}
+
+// ImagePattern defines a pattern and its associated authorties
+// If multiple patterns match a particular image, then ALL of
+// those authorities must be satisfied for the image to be admitted.
+type ImagePattern struct {
+	// +optional
```
[the remark this answers is not in the document]
Comment (vaikas) [reply pkg/apis/cosigned/v1beta1/clusterimagepolicy_types.go:68]: @elfotografo007 Thanks! Nope, left over from when we removed regex.

**Review by hectorj2f — APPROVED** [review]
LGTM, it looks good to me. We can add the e2e tests in a different PR.

**Comment by hectorj2f** [comment]
thanks @dlorenc :)

**Pull request merged**